types/producer.d.ts

```diff
@@ -1,11 +1,14 @@
-export interface Options {
+export interface ProduceOptions {
     readonly onSuccess?: Function,
     readonly onError?: Function
 }
 
+export type Item = string | object;
+export type Message = Item | Array<Item>;

```

Comment:
Move to a separate file, there's not a great reason why this should belong to producer. This way, you avoid having `base` depend on `producer`, which is odd. If anything, it would be the other way around.

Reply:
Base is still going to depend on a `message.d.ts` file for instance.

Reply:
I was imagining `Message` belonging to `base`, but it's 👍 as it is.